testslide/matchers.py

```diff
@@ -0,0 +1,309 @@
+#!/usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+import re
+from typing import List, Dict
+
+class AlreadyChainedException(Exception):
+    pass
```

Comment:
It is important to add a ``__str__`` here, so when users get this exception, they get a short message explaining what happened, and that's not exactly their fault.